### PR TITLE
Issue #271 respect accept header

### LINKED ISSUE #271: Accept header values not being considered when generating a response
**Describe the bug**

**To Reproduce**

1. create the following application

```typescript
// File: app.ts

import { Drash } from "https://deno.land/x/drash/mod.ts";

class HomeResource extends Drash.Http.Resource {
  static paths = ["/"];
  public GET() {
    this.response.body = "Hello";
    return this.response;
  }
}

const server = new Drash.Http.Server({
  logger: new Drash.CoreLoggers.ConsoleLogger({
    enabled: true,
    level: "debug",
  }),
  resources: [HomeResource],
});

server.run({
  hostname: "localhost",
  port: 1447,
});

console.log("Server listening: http://localhost:1447");

```

2. run the app
3. Go to localhost:1447 in insomnia or postman
4. Send a GET request (expected response content-type header should be application/json)
5. Send a GET request with the Accept header of text/html (expected content-type header should be text/html, but it's not)

**Expected behavior**

The Accept header should be considered when generating a response. The response object sets its content-type header in it's `constructor()`. Currently, it still uses the `response_content_type` code that was removed in an earlier version. This code needs to be removed and replaced with the code that parses the Accept header.

### PR DESCRIPTION
**Pull request by crookse**
Fixes #271 

This PR does the following:

* moves accepts() from HttpRequestService to HttpService
* removes logic around `response_content_type` and respects the `accept` header
* removes tests for `response_content_type`

### CODE REVIEW

File: src/http/response.ts [hunk 1]
```diff
@@ -74,15 +82,46 @@ export class Response {
...
     this.headers = new Headers();
     this.template_engine = options.template_engine;
     this.views_path = options.views_path;
-    this.headers.set("Content-Type", request.response_content_type);
+    this.headers.set("Content-Type", this.getContentType());
   }
 
   // FILE MARKER: METHODS - PUBLIC /////////////////////////////////////////////
 
+  protected getContentType(): string {
```
Comment (ebebbington) [inline src/http/response.ts:95]: We able to have a doc block for this?

**Comment by ebebbington** [comment]
I like what you've done though, how you've written the logic for the getContentType method

File: src/http/response.ts (hunk 1 above)
Comment (crookse) [reply src/http/response.ts]: done!

**Comment by crookse** [comment]
> Get the content type from the request objects "Accept" header

mad the change. yea i should've been more verbose here. also changed the method name

**Comment by crookse** [comment]
> Maybe add: `If no default response content type was passed in, then defaults to application/json`, and also something like `Returns the first accepted content type if multiple as within the accept header` somewhere?

made the change

**Pull request merged**